src/Native/System.Private.CoreLib.Native/pal_path.cpp

```diff
@@ -0,0 +1,67 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include "pal_common.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#if defined(__APPLE__)
+#include <mach-o/dyld.h>
+#endif
+
+#if defined(__linux__)
+#define symlinkEntrypointExecutable "/proc/self/exe"
+#endif
+
+// Get full path to the executable for the current process resolving symbolic links.
+// On success, the function returns the size of the buffer required to hold the result.
+// In case of an error, the function returns -1.
+extern "C" int32_t CoreLibNative_GetExecutableAbsolutePath(char* buffer, int32_t bufferSize)
```

Comment:
I only now noticed that a similar function is being added in https://github.com/dotnet/corert/pull/3571 :)